### PR TITLE
LCAM 925 MAAT Specific Retrieve Hardship Endpoint

### PR DESCRIPTION
**Pull request by nigelpain**
## What

[Link to story](https://dsdmoj.atlassian.net/jira/software/projects/LCAM/boards/881?selectedIssue=LCAM-925)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/staticdata/enums/HardshipReviewDetailCode.java [hunk 1]
```diff
@@ -26,34 +28,26 @@ public enum HardshipReviewDetailCode {
...
     PENSION_PAY("PENSION PAY", "Pension Payments", HardshipReviewDetailType.EXPENDITURE.getType()),
-    MEDICAL_COSTS("MEDICAL COSTS", "Medical Costs", HardshipReviewDetailType.EXPENDITURE.getType()),
+    // Typo in the data that is already in the table
+    MEDICAL_COSTS("MEDIAL COSTS", "Medical Costs", HardshipReviewDetailType.EXPENDITURE.getType()),
     OTHER("OTHER", "Other", HardshipReviewDetailType.EXPENDITURE.getType()),
     MEDICAL_GROUNDS("MEDICAL GROUNDS", "Medical Grounds", HardshipReviewDetailType.INCOME.getType()),
     SUSPENDED_WORK("SUSPENDED WORK", "Suspended from work", HardshipReviewDetailType.INCOME.getType()),
     OTHER_INC("OTHER INC", "Other", HardshipReviewDetailType.INCOME.getType());
 
     @JsonPropertyDescription("Hardship review detail codes that are valid")
```
Comment (skasthuri36326) [inline crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/staticdata/enums/HardshipReviewDetailCode.java:38]: ExtraExpenditureDetailCode and HardshipReviewDetailCode seems to be identical. Can we have a base Enum?

Comment (Mhowell494) [reply crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/staticdata/enums/HardshipReviewDetailCode.java:38]: I believe there are a few extra fields in the `HardshipReviewDetailCode` enum. However, I had planned to create an extra ticket to do some cleanup. The second field type is never used and there is a missing value thats in the DB but not the enum.

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/controller/HardshipController.java [hunk 2]
```diff
@@ -64,6 +65,27 @@ public ResponseEntity<ApiCalculateHardshipByDetailResponse> calculateHardshipFor
...
+                    schema = @Schema(implementation = ApiPerformHardshipResponse.class)
+            )
+    )
+    @DefaultHTTPErrorResponse
+    public ResponseEntity<ApiFindHardshipResponse> find(
+            @PathVariable int hardshipReviewId,
+            @Parameter(description = "Used to trace calls between services")
+            @RequestHeader(value = "Laa-Transaction-Id", required = false) String laaTransactionId) {
+
+        MDC.put("laaTransactionId", laaTransactionId);
```
[the remark this answers is not in the document]
Comment (Mhowell494) [reply crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/controller/HardshipController.java:82]: Have created a separate ticket in the backlog to tackle this for all of our services: https://dsdmoj.atlassian.net/browse/LCAM-994

**Review by skasthuri36326 — APPROVED** [review]
Looks fine to me but the PR is marked as WIP.

**Comment by Mhowell494** [comment]
> Looks fine to me but the PR is marked as WIP.

Good point! I forgot to remove that :)

**Pull request merged**